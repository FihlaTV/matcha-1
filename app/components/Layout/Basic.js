import React, { type Node } from 'react';
import { connect } from 'react-redux';
import { StatusBar, View } from 'react-native';
import Header from 'app/components/Header';
import Nav from 'app/components/Nav';
import Toast from 'app/components/Layout/Toast';
import Loader from 'app/components/Layout/Loader';
import { selectToastIsShow, selectToastMessage, selectIsLoading } from 'app/components/Layout/selector';

type Props = {|
  children: Node,
  hasBack?: boolean,
  headerActions?: Array<Node>,
  isLoading: boolean,
  noTabs?: boolean,
  title?: string,
|};

const Layout = (props: Props) => (
  <View style={{ flex: 1 }}>
    <StatusBar />
    <Header hasBack={props.hasBack} title={props.title} actions={props.headerActions} />
    <View style={{ flex: 1 }}>
      {props.children}
    </View>
    { !props.noTabs && <Nav />}
    <Toast
      message={props.toastMessage}
      isShow={props.toastIsShow}
    />
    <Loader isLoading={props.isLoading} />
  </View>
);

Layout.defaultProps = {
  hasBack: false,
  headerActions: [],
  noTabs: false,
  title: undefined,
};

const mapStateToProps = (state) => ({
  isLoading: selectIsLoading(state),
  toastIsShow: selectToastIsShow(state),
  toastMessage: selectToastMessage(state),
});

export default connect(mapStateToProps)(Layout);
