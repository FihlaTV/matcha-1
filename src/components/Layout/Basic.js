import React, { type Node } from 'react';
import { StatusBar, View } from 'react-native';
import Header from 'app/components/Header';
import Nav from 'app/components/Nav';

type Props = {|
  children: Node,
  noTabs?: boolean,
  hasBack?: boolean,
  title?: string,
  headerActions?: Array<Node>,
|};

const BasicLayout = (props: Props) => (
  <View style={{ flex: 1 }}>
    <StatusBar />
    <Header hasBack={props.hasBack} title={props.title} actions={props.headerActions} />
    <View style={{ flex: 1 }}>
      {props.children}
    </View>
    { !props.noTabs && <Nav />}
  </View>
);

BasicLayout.defaultProps = {
  hasBack: false,
  headerActions: [],
  noTabs: false,
  title: undefined,
};

export default BasicLayout;
