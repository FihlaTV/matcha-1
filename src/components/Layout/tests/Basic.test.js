import React from 'react';
import Basic from 'src/components/Layout/Basic';
import BasicWeb from 'src/components/Layout/Basic.web';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Layout', () => {
  describe('render test', () => {
    it('should render', () => {
      const wrapper = shallow(<Basic />);
      expect(toJson(wrapper)).toMatchSnapshot();
      const wrapperWeb = shallow(<BasicWeb />);
      expect(toJson(wrapperWeb)).toMatchSnapshot();
    });
  });
});
