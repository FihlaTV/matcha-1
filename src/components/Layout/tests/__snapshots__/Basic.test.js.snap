// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render test should render 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    showHideTransition="fade"
  />
  <InjectIntl(Header)
    actions={Array []}
    hasBack={false}
  />
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <InjectIntl(Nav) />
</Component>
`;

exports[`Layout render test should render 2`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    showHideTransition="fade"
  />
  <InjectIntl(Nav) />
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {},
      ]
    }
  />
  <InjectIntl(Footer) />
</Component>
`;
